### PR TITLE
prb/updater: more option

### PR DESCRIPTION
**Pull request by piotrbartman**
add option needed by gui

### CODE REVIEW

File: vmupdate/vmupdate.py [hunk 1]
```diff
@@ -90,17 +111,124 @@ def get_targets(args, app):
...
+                      if vm.klass in ('AppVM', 'DispVM')
+                      and vm.is_running()}
+        print("Following qubes CAN be restarted:",
+              ",".join((target.name for target in to_restart)))
+        return
+
+    # first shutdown templates to apply changes to the root volume
+    for template in templates_to_shutdown:
+        try:
+            template.shutdown()
```
Comment (marmarek) [inline vmupdate/vmupdate.py:188]: you need to wait for shutdown to complete before restarting appvms. Take a look at https://github.com/QubesOS/qubes-core-admin-client/blob/main/qubesadmin/events/utils.py#L42 for less busy-wait approach.

File: vmupdate/vmupdate.py [hunk 2]
```diff
@@ -90,17 +111,124 @@ def get_targets(args, app):
...
+    for vm in to_restart:
+        if next(vm.connected_vms(), None) is None:
+            try:
+                vm.shutdown()
+                wait_for.append(vm)
+            except qubesadmin.exc.QubesVMError as exc:
+                print(exc, file=sys.stderr)
+        else:
+            try:
+                vm.kill()
```
Comment (marmarek) [inline vmupdate/vmupdate.py:216]: should be `vm.shutdown(force=True)` - it does graceful shutdown, but ignores dependencies check (like networking). In fact, you can always do `vm.shutdown(force=True)` (as you're going to restart it back) and don't need to check connected_vms()

File: vmupdate/vmupdate.py [hunk 3]
```diff
@@ -90,17 +111,124 @@ def get_targets(args, app):
...
+                wait_for.append(vm)
+            except qubesadmin.exc.QubesVMError as exc:
+                print(exc, file=sys.stderr)
+
+    # wait for qubes to shut down
+    while any(vm.is_running() for vm in wait_for):
+        time.sleep(1)
```
Comment (marmarek) [inline vmupdate/vmupdate.py:223]: see the other comment about waiting for shutdown

File: vmupdate/vmupdate.py [hunk 4]
```diff
@@ -50,14 +59,26 @@ def parse_args(args):
     group.add_argument('--all', action='store_true',
                        help='Target all non-disposable VMs (TemplateVMs and '
                             'AppVMs)')
-
+    group.add_argument('--n', action='store',
```
Comment (marmarek) [inline vmupdate/vmupdate.py:62]: `--n` looks weird, I liked `--smart` more, but maybe @marmarta has some better idea?

**Review by marmarek — CHANGES_REQUESTED** [review]

File: vmupdate/vmupdate.py (hunk 4 above)
Comment (piotrbartman) [reply vmupdate/vmupdate.py:62]: yes I also like `qubes-vm-update --smart` but `qubes-vm-update --smart 5` doesn't make sense to me. For now smart behavior is default so `qubes-vm-update` will do what we want, so first option is out. 
I'm open to proposition for a parameter name "how long without checking for updates is unacceptable"

Comment (marmarta) [reply vmupdate/vmupdate.py:62]: `--smart-time`, `--check-older`?

Comment (marmarta) [reply vmupdate/vmupdate.py:62]: Also chatGPT proposed '--update-if-stale N', and frankly it sounds like a good idea

**Review by marmarek — CHANGES_REQUESTED** [review]

**Comment by marmarek** [comment]
Most recent test run:
```
Traceback (most recent call last):
  File "/usr/lib64/python3.11/contextlib.py", line 81, in inner
    return func(*args, **kwds)
           ^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/qubes/tests/integ/vm_update.py", line 626, in test_111_update_via_proxy_qubes_vm_update_cli
    self.update_via_proxy_qubes_vm_update_impl(
  File "/usr/lib/python3.11/site-packages/qubes/tests/integ/vm_update.py", line 532, in update_via_proxy_qubes_vm_update_impl
    self.run_qubes_vm_update_and_assert(
  File "/usr/lib/python3.11/site-packages/qubes/tests/integ/vm_update.py", line 471, in run_qubes_vm_update_and_assert
    self.fail(
AssertionError: qubes-vm-update return unexpected code: 1 in [0]
test-inst-vm1: ERROR (exception 'NoneType' object has no attribute 'put')
```

**Review by marmarek — APPROVED** [review]

**Pull request merged**